worker/modelcache/worker.go

```diff
@@ -21,25 +21,33 @@ import (
 	"github.com/juju/juju/state/multiwatcher"
 )
 
+type BackingWatcher interface {
+	Next() ([]multiwatcher.Delta, error)
+	Stop() error
+}
+
 // Config describes the necessary fields for NewWorker.
 type Config struct {
 	Logger               Logger
-	StatePool            *state.StatePool
 	PrometheusRegisterer prometheus.Registerer
 	Cleanup              func()
+
 	// Notify is used primarily for testing, and is passed through
 	// to the cache.Controller. It is called every time the controller
 	// processes an event.
 	Notify func(interface{})
+
+	// WatchFactory supplies the
+	WatcherFactory func() BackingWatcher
```

Comment:
Instead of passing the state pool, it is closed over to create this multi-watcher-returning factory.